eng/Subsets.props

```diff
@@ -76,6 +77,12 @@
     <_subset>-$(_subset)-</_subset>
   </PropertyGroup>
 
+  <PropertyGroup>
+    <RuntimeFlavor Condition="'$(TargetOS)' == 'iOS'">Mono</RuntimeFlavor>
+    <RuntimeFlavor Condition="'$(RuntimeFlavor)' == '' and $(_subsetCategory.Contains('mono')) and !$(_subsetCategory.Contains('coreclr'))">Mono</RuntimeFlavor>
+    <RuntimeFlavor Condition="'$(RuntimeFlavor)' == ''">CoreCLR</RuntimeFlavor>
+  </PropertyGroup>
```

Comment:
This feels a bit wrong to set in Subsets.props but I need access to `_subsetCategory`.
A better approach might be to set two new properties `BuildIncludesCoreCLR` and `BuildIncludesMono` in here and then do the RuntimeFlavor logic in Directory.Build.props based off of that.

@ViktorHofer what do you think?

Reply:
I think it would definitely be better to set some properties to know which components of the build are being built and then condition based on that in `Directory.Build.props`. I don't like the fact of having multiple places calculating `RuntimeFlavor` seems less maintainable and more error prone. We could always use `_subsetCategory` outside of this file, but it seems cleaner to not use private intended properties outside.

Reply:
I'm ok with deriving `RuntimeFlavor` from the chosen `SubsetCategory`. Giving it a second though, I'm even ok with setting this here...

> I don't like the fact of having multiple places calculating RuntimeFlavor

RuntimeFlavor isn't calculated anywhere, it's only passed in into the entrypoint scripts.

Reply:
> RuntimeFlavor isn't calculated anywhere, it's only passed in into the entrypoint scripts.

Oh true, I thought it was already calculated in  Directory.Build.props, that's why I thought it was better to do it there, but you're right, it is not. So then it feels like this is correct as is.